feat(event_handler): Add new EventHandler for Async Lambda

sinofseven opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
**Issue number:** #5627 

## Summary

Add `AsyncTriggerResolver`.

### Changes

> Please provide a summary of what's being changed

Added EventHandler for asynchronous Lambda execution.

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [ ] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [ ] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented
* [ ] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

sinofseven committed:
impl

sinofseven committed:
fix mypy, and test aws_config_rule.py

sinofseven committed:
fix SonarCloud Code Analysis on sns.py

sinofseven committed:
fix import order

sinofseven committed:
add test for cloud_watch_alarm.py

sinofseven committed:
add test for cloud watch logs

sinofseven committed:
add test for code_deploy_lifecycle_hook.py

sinofseven committed:
add test for event_bridge.py

sinofseven committed:
fix S3Route

anafalcao commented:
Hi @sinofseven! Thank you so much for opening this PR. Do you need any help here?

sinofseven committed:
fix unit test for aws_config_fule.py

sinofseven committed:
fix implement for cloud_watch_alarm.py

sinofseven committed:
fix implement for cloud_watch_logs.py

sinofseven committed:
fix implement for event_bridge.py

sinofseven committed:
fix implement for ses.py

sinofseven committed:
fix test for cloud_watch_alarm.py

sinofseven committed:
fix test for aws_config_rule.py

sinofseven commented:
Hi, @anafalcao!

Thanks very much for your suggestion!

I have a few problems at the moment.
I am hoping to discuss these with you.

- What must be done in tests other than unit tests
- Regarding the test case for the unit test.

## What must be done in tests other than unit tests
 
For the EventHandler implementation I am creating, I believe that the event samples provided for unit testing of data_classes are sufficient for testing.

However, the EventHandler test for API Gateway implements E2E testing.
I am wondering if it is necessary to implement E2E tests for the EventHandler I am creating.

## Regarding the test case for the unit test

The unit test attempted to check all cases, including correct and incorrect values for all arguments of each Route class, as well as the presence or absence of arguments.

Therefore, in creating the test, I wrote all the patterns of the test case in a Google Spreadsheet.

https://docs.google.com/spreadsheets/d/1eioW7rALlFdkv2Q0y8ERZnMnJuJK3QWoPfhoJMTlqaA/edit?usp=sharing

Following this, I implemented a unit test in `tests/unit/event_handler/_async_execution/_routes/test_aws_config_rule.py`.
(The other tests were created before we properly managed the test cases.)

Here is where I ran into a problem: I had a very large number of test cases in having many arguments Route class.
(For example, we found that the `S3Route` class with 9 arguments requires just under 20,000 test cases.)

Therefore, I am wondering how far to describe the test cases.

sinofseven committed:
fix implement for sns.py

sinofseven committed:
add test for cloud_watch_alarm.py

leandrodamascena commented:
Hi @sinofseven, first of all, sorry for the late reply, but I was super busy these days. I was reading the code and it covers most of the aspects we expect from this utility, such as: routing per key, detail, object, event name and others. I must say we have a SUPER SUPER nice work here, thank you very much for that.

I have a proposal to follow with this PR and please let me know what do you think.

### Creating a RFC

I'm going to start an RFC with what this implementation should cover. Powertools is available in [4 different languages](https://github.com/aws-powertools/) and the idea is that this RFC can cover the design of this implementation, features that will be supported and resolvers that we will support in the first version. Don't worry we have people to implement in different languages. I need a week to have a v1 of this RFC and then you can contribute.

### Keeping this PR open

I will keep this PR open while we discuss the RFC, which will probably take something like 1 month, because we need to hear from the community, from you, from the other maintainers and we need to agree on the implementation. I will also run local tests with this PR to get more insight into the codebase, changes we need to make and other things.

### Integrations

We have a utility called EventSource Data Class, where we map the event that comes from any service to Lambda, we can integrate this utility with this one and create an experience like:

```python
from aws_lambda_powertools.async_resolver import S3Resolver

app = S3Resolver()

@app.resolve_by_name(object_name="powertools.txt")
def resolve_by_name():
    print(f"Event time {app.event_time}") # We can access the event properties
```

We also have a utility called Parser where we use Pydantic templates to do data parsing and validation against S3 Events. We can think of an idea how to integrate with that.

### My personal opinion

While I like the idea of ​​having multiple resolvers like S3, EventBridge, SNS, SES, it might make sense to step back and agree to the following:

1 - Make the base router strong and consistent enough to support many types of resolvers
2 - Implement 2 initial resolvers like S3 and EventBridger, because I think events in this case are simpler to interpret and route
3 - Create tests that can be reused for many resolvers.
4 - As soon as we validate the 2 initial resolvers, we can add more.

I would love to hear your feedback and work together on this RFC to design this utility API and add this feature to Powertools.

sinofseven committed:
add test for other event file

sinofseven committed:
fix test for cloud_watch_alarm.py

sinofseven committed:
fix test for cloud_watch_logs

sinofseven committed:
fix test for code_deploy_lifecycle_hook.py

sinofseven committed:
fix test for event_bridge.py

leandrodamascena committed:
Merge branch 'develop' into impl-prototype-async-execution-resolver

leandrodamascena committed:
chore(ci): adding poetry export plugin to support v2 (#5941)

Adding export plugin

github-actions[bot] committed:
chore(ci): bump version to 3.5.0 (#5942)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

anafalcao committed:
fix(docs): typo in a service name in Event Handler (#5944)

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a0 (#5945)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#5946)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
fix(logger): child logger must respect log level (#5950)

Child logger must respect log level

leandrodamascena committed:
docs(layer): update layer version number - v3.5.0 (#5952)

Updating layer version

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a1 (#5954)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#5955)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump actions/setup-python from 5.3.0 to 5.4.0 (#5960)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.3.0 to 5.4.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/0b93645e9fea7318ecaed2b359559ac225c90a2b...42375524e23c412d93fb67b49958b491fce71c38)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

leandrodamascena committed:
chore(ci): install & configure mkdocstrings plugin (#5959)

* Installing and configuring  mkdocstrings plugin

* Installing and configuring  mkdocstrings plugin

---------

Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps): bump docker/setup-qemu-action from 3.2.0 to 3.3.0 (#5961)

Bumps [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) from 3.2.0 to 3.3.0.
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/49b3bc8e6bdd4a60e6116a5414239cba5943d3cf...53851d14592bedcffcf25ea515637cff71ef929a)

---
updated-dependencies:
- dependency-name: docker/setup-qemu-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump codecov/codecov-action from 5.1.2 to 5.3.1 (#5964)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from 5.1.2 to 5.3.1.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Changelog](https://github.com/codecov/codecov-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/codecov/codecov-action/compare/1e68e06f1dbfde0e4cefc87efeba9e4643565303...13ce06bfc6bbe3ecf90edbbf1bc32fe5978ca1d3)

---
updated-dependencies:
- dependency-name: codecov/codecov-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump mkdocs-material from 9.5.50 to 9.6.1 (#5966)

Bumps [mkdocs-material](https://github.com/squidfunk/mkdocs-material) from 9.5.50 to 9.6.1.
- [Release notes](https://github.com/squidfunk/mkdocs-material/releases)
- [Changelog](https://github.com/squidfunk/mkdocs-material/blob/master/CHANGELOG)
- [Commits](https://github.com/squidfunk/mkdocs-material/compare/9.5.50...9.6.1)

---
updated-dependencies:
- dependency-name: mkdocs-material
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps): bump actions/setup-node from 4.1.0 to 4.2.0 (#5963)

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4.1.0 to 4.2.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/39370e3970a6d050c480ffad4ff0ed4d3fdee5af...1d0ff469b7ec7b3cb9d8673fde0c81c44821de2a)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact from 4.5.0 to 4.6.0 (#5962)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 4.5.0 to 4.6.0.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/6f51ac03b9356f520e9adb1b1b7802705f340c2b...65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump isort from 5.13.2 to 6.0.0 (#5965)

Bumps [isort](https://github.com/pycqa/isort) from 5.13.2 to 6.0.0.
- [Release notes](https://github.com/pycqa/isort/releases)
- [Changelog](https://github.com/PyCQA/isort/blob/main/CHANGELOG.md)
- [Commits](https://github.com/pycqa/isort/compare/5.13.2...6.0.0)

---
updated-dependencies:
- dependency-name: isort
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a2 (#5970)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump cfn-lint from 1.22.7 to 1.23.1 (#5967)

Bumps [cfn-lint](https://github.com/aws-cloudformation/cfn-lint) from 1.22.7 to 1.23.1.
- [Release notes](https://github.com/aws-cloudformation/cfn-lint/releases)
- [Changelog](https://github.com/aws-cloudformation/cfn-lint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws-cloudformation/cfn-lint/compare/v1.22.7...v1.23.1)

---
updated-dependencies:
- dependency-name: cfn-lint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): changelog rebuild (#5971)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump ruff from 0.9.3 to 0.9.4 (#5969)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.3 to 0.9.4.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.3...0.9.4)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump black from 24.10.0 to 25.1.0 (#5968)

Bumps [black](https://github.com/psf/black) from 24.10.0 to 25.1.0.
- [Release notes](https://github.com/psf/black/releases)
- [Changelog](https://github.com/psf/black/blob/main/CHANGES.md)
- [Commits](https://github.com/psf/black/compare/24.10.0...25.1.0)

---
updated-dependencies:
- dependency-name: black
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

leandrodamascena committed:
docs(api): migrating the validation utility to mkdocstrings (#5972)

Mkdocstrings - Validation module

anafalcao committed:
feat(logger): add clear_state method (#5956)

* feat(logger): add clear_state method to Logger

* changes after feedback

* change reset default keys

---------

Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps): bump aws-actions/configure-aws-credentials from 4.0.2 to 4.0.3 (#5975)

chore(deps): bump aws-actions/configure-aws-credentials

Bumps [aws-actions/configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials) from 4.0.2 to 4.0.3.
- [Release notes](https://github.com/aws-actions/configure-aws-credentials/releases)
- [Changelog](https://github.com/aws-actions/configure-aws-credentials/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws-actions/configure-aws-credentials/compare/e3dd6a429d7300a6a4c196c26e071d42e0343502...4fc4975a852c8cd99761e2de1f4ba73402e44dd9)

---
updated-dependencies:
- dependency-name: aws-actions/configure-aws-credentials
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps): bump release-drafter/release-drafter from 6.0.0 to 6.1.0 (#5976)

dependabot[bot] committed:
chore(deps): bump zgosalvez/github-actions-ensure-sha-pinned-actions from 3.0.18 to 3.0.20 (#5977)

dependabot[bot] committed:
chore(deps): bump pypa/gh-action-pypi-publish from 1.12.3 to 1.12.4 (#5980)

dependabot[bot] committed:
chore(deps): bump actions/setup-go from 5.2.0 to 5.3.0 (#5978)

dependabot[bot] committed:
chore(deps): bump squidfunk/mkdocs-material from `41942f7` to `471695f` in /docs (#5979)

github-actions[bot] committed:
chore(ci): changelog rebuild (#5981)

github-actions[bot] committed:
chore(ci): changelog rebuild (#5982)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a3 (#5998)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#5999)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
docs(api): migrating the typing utility to mkdocstrings (#5996)

Mkdocstrings - Typing module

Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

leandrodamascena committed:
refactor(metrics): Improve type annotations for metrics decorator (#6000)

Fix type annotation with metrics decorator

leandrodamascena committed:
docs(api): migrating the streaming utility to mkdocstrings (#6003)

Mkdocstrings - Streaming module

leandrodamascena committed:
docs(api): migrating the parser utility to mkdocstrings (#6004)

* Mkdocstrings - Parser module

* Mkdocstrings - Parser module

leandrodamascena committed:
docs(api): migrating the parameters utility to mkdocstrings (#6005)

Mkdocstrings - Parameters  module

leandrodamascena committed:
docs(api): migrating the jmespath utility to mkdocstrings (#6006)

Mkdocstrings - JMESPath module

leandrodamascena committed:
docs(api): migrating the idempotency utility to mkdocstrings (#6007)

Mkdocstrings - Idempotency module

leandrodamascena committed:
docs(api): migrating the AppConfig utility to mkdocstrings (#6008)

Mkdocstrings - AppConfig module

dependabot[bot] committed:
chore(deps): bump squidfunk/mkdocs-material from `471695f` to `7e841df` in /docs (#6012)

chore(deps): bump squidfunk/mkdocs-material in /docs

Bumps squidfunk/mkdocs-material from `471695f` to `7e841df`.

---
updated-dependencies:
- dependency-name: squidfunk/mkdocs-material
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump mkdocstrings-python from 1.13.0 to 1.14.2 (#6011)

Bumps [mkdocstrings-python](https://github.com/mkdocstrings/python) from 1.13.0 to 1.14.2.
- [Release notes](https://github.com/mkdocstrings/python/releases)
- [Changelog](https://github.com/mkdocstrings/python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mkdocstrings/python/compare/1.13.0...1.14.2)

---
updated-dependencies:
- dependency-name: mkdocstrings-python
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump mkdocs-material from 9.6.1 to 9.6.2 (#6009)

Bumps [mkdocs-material](https://github.com/squidfunk/mkdocs-material) from 9.6.1 to 9.6.2.
- [Release notes](https://github.com/squidfunk/mkdocs-material/releases)
- [Changelog](https://github.com/squidfunk/mkdocs-material/blob/master/CHANGELOG)
- [Commits](https://github.com/squidfunk/mkdocs-material/compare/9.6.1...9.6.2)

---
updated-dependencies:
- dependency-name: mkdocs-material
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.10 to 1.36.11 (#6010)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.10 to 1.36.11.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

leandrodamascena committed:
docs(api): migrating the data masking utility to mkdocstrings (#6013)

Mkdocstrings - DataMasking module

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.10 to 1.36.12 (#6014)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.10 to 1.36.12.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

leandrodamascena committed:
docs(api): migrating the event source data classes utility to mkdocstrings (#6015)

Mkdocstrings - Data Classes module

philiptzou committed:
feat(openapi): enhance support for tuple return type validation (#5997)

Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

leandrodamascena committed:
docs(api): migrating the batch utility to mkdocstrings (#6016)

* Mkdocstrings - Batch module

* Mkdocstrings - Batch module

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a4 (#6018)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
docs(api): migrating the tracer utility to mkdocstrings (#6017)

Mkdocstrings - Tracer

github-actions[bot] committed:
chore(ci): changelog rebuild (#6020)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
docs(api): migrating the Middleware Factory utility to mkdocstrings (#6019)

* Mkdocstrings - Middleware Factory

* Mkdocstrings - Middleware Factory

---------

Co-authored-by: Simon Thulbourn <sthulb@users.noreply.github.com>

leandrodamascena committed:
docs(api): migrating the logger utility to mkdocstrings (#6021)

Mkdocstrings - Logging

leandrodamascena committed:
docs(api): migrating the metrics utility to mkdocstrings (#6022)

Mkdocstrings - Metrics

leandrodamascena committed:
docs(api): migrating the event handler utility to mkdocstrings (#6023)

Mkdocstrings - Event handler

leandrodamascena committed:
chore(ci): remove pdoc3 library (#6024)

* Removing pdoc3

* Removing pdoc3

* Removing pdoc3

* Removing pdoc3

* Removing pdoc3

dependabot[bot] committed:
chore(deps-dev): bump mkdocstrings-python from 1.14.2 to 1.14.4 (#6025)

Bumps [mkdocstrings-python](https://github.com/mkdocstrings/python) from 1.14.2 to 1.14.4.
- [Release notes](https://github.com/mkdocstrings/python/releases)
- [Changelog](https://github.com/mkdocstrings/python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mkdocstrings/python/compare/1.14.2...1.14.4)

---
updated-dependencies:
- dependency-name: mkdocstrings-python
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a5 (#6026)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
chore(ci): add new script to bump Lambda layer version (#6001)

* Adding new script to bump layer version

* Adding new script to bump layer version

* Making changes to work with GNU sed

github-actions[bot] committed:
chore(ci): changelog rebuild (#6027)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a6 (#6033)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump mkdocstrings-python from 1.14.4 to 1.14.5 (#6032)

Bumps [mkdocstrings-python](https://github.com/mkdocstrings/python) from 1.14.4 to 1.14.5.
- [Release notes](https://github.com/mkdocstrings/python/releases)
- [Changelog](https://github.com/mkdocstrings/python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mkdocstrings/python/compare/1.14.4...1.14.5)

---
updated-dependencies:
- dependency-name: mkdocstrings-python
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump cfn-lint from 1.23.1 to 1.24.0 (#6030)

Bumps [cfn-lint](https://github.com/aws-cloudformation/cfn-lint) from 1.23.1 to 1.24.0.
- [Release notes](https://github.com/aws-cloudformation/cfn-lint/releases)
- [Changelog](https://github.com/aws-cloudformation/cfn-lint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws-cloudformation/cfn-lint/compare/v1.23.1...v1.24.0)

---
updated-dependencies:
- dependency-name: cfn-lint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump cdklabs-generative-ai-cdk-constructs from 0.1.290 to 0.1.291 (#6031)

chore(deps-dev): bump cdklabs-generative-ai-cdk-constructs

Bumps [cdklabs-generative-ai-cdk-constructs](https://github.com/awslabs/generative-ai-cdk-constructs) from 0.1.290 to 0.1.291.
- [Release notes](https://github.com/awslabs/generative-ai-cdk-constructs/releases)
- [Changelog](https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/awslabs/generative-ai-cdk-constructs/compare/v0.1.290...v0.1.291)

---
updated-dependencies:
- dependency-name: cdklabs-generative-ai-cdk-constructs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.12 to 1.36.14 (#6029)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.12 to 1.36.14.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump mypy from 1.14.1 to 1.15.0 (#6028)

Bumps [mypy](https://github.com/python/mypy) from 1.14.1 to 1.15.0.
- [Changelog](https://github.com/python/mypy/blob/master/CHANGELOG.md)
- [Commits](https://github.com/python/mypy/compare/v1.14.1...v1.15.0)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6034)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

dreamorosi committed:
chore(docs): enable privacy plugin in docs (#6036)

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk-lib from 2.177.0 to 2.178.0 (#6038)

Bumps [aws-cdk-lib](https://github.com/aws/aws-cdk) from 2.177.0 to 2.178.0.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/compare/v2.177.0...v2.178.0)

---
updated-dependencies:
- dependency-name: aws-cdk-lib
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump ruff from 0.9.4 to 0.9.5 (#6039)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.4 to 0.9.5.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.4...0.9.5)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk-aws-lambda-python-alpha from 2.177.0a0 to 2.178.0a0 (#6041)

chore(deps-dev): bump aws-cdk-aws-lambda-python-alpha

Bumps [aws-cdk-aws-lambda-python-alpha](https://github.com/aws/aws-cdk) from 2.177.0a0 to 2.178.0a0.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/commits)

---
updated-dependencies:
- dependency-name: aws-cdk-aws-lambda-python-alpha
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk from 2.177.0 to 2.178.0 (#6040)

Bumps [aws-cdk](https://github.com/aws/aws-cdk/tree/HEAD/packages/aws-cdk) from 2.177.0 to 2.178.0.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/commits/v2.178.0/packages/aws-cdk)

---
updated-dependencies:
- dependency-name: aws-cdk
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps): bump docker/setup-buildx-action from 3.8.0 to 3.9.0 (#6042)

Bumps [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) from 3.8.0 to 3.9.0.
- [Release notes](https://github.com/docker/setup-buildx-action/releases)
- [Commits](https://github.com/docker/setup-buildx-action/compare/6524bf65af31da8d45b59e8c27de4bd072b392f5...f7ce87c1d6bead3e36075b2ce75da1f6cc28aaca)

---
updated-dependencies:
- dependency-name: docker/setup-buildx-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps): bump docker/setup-qemu-action from 3.3.0 to 3.4.0 (#6043)

Bumps [docker/setup-qemu-action](https://github.com/docker/setup-qemu-action) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/docker/setup-qemu-action/releases)
- [Commits](https://github.com/docker/setup-qemu-action/compare/53851d14592bedcffcf25ea515637cff71ef929a...4574d27a4764455b42196d70a065bc6853246a25)

---
updated-dependencies:
- dependency-name: docker/setup-qemu-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a7 (#6044)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6045)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk from 2.178.0 to 2.178.1 (#6053)

Bumps [aws-cdk](https://github.com/aws/aws-cdk/tree/HEAD/packages/aws-cdk) from 2.178.0 to 2.178.1.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/commits/v2.178.1/packages/aws-cdk)

---
updated-dependencies:
- dependency-name: aws-cdk
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk-lib from 2.178.0 to 2.178.1 (#6047)

Bumps [aws-cdk-lib](https://github.com/aws/aws-cdk) from 2.178.0 to 2.178.1.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/compare/v2.178.0...v2.178.1)

---
updated-dependencies:
- dependency-name: aws-cdk-lib
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk-aws-lambda-python-alpha from 2.178.0a0 to 2.178.1a0 (#6048)

chore(deps-dev): bump aws-cdk-aws-lambda-python-alpha

Bumps [aws-cdk-aws-lambda-python-alpha](https://github.com/aws/aws-cdk) from 2.178.0a0 to 2.178.1a0.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/main/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/commits)

---
updated-dependencies:
- dependency-name: aws-cdk-aws-lambda-python-alpha
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.14 to 1.36.15 (#6049)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.14 to 1.36.15.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump cdklabs-generative-ai-cdk-constructs from 0.1.291 to 0.1.292 (#6051)

chore(deps-dev): bump cdklabs-generative-ai-cdk-constructs

Bumps [cdklabs-generative-ai-cdk-constructs](https://github.com/awslabs/generative-ai-cdk-constructs) from 0.1.291 to 0.1.292.
- [Release notes](https://github.com/awslabs/generative-ai-cdk-constructs/releases)
- [Changelog](https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/awslabs/generative-ai-cdk-constructs/compare/v0.1.291...v0.1.292)

---
updated-dependencies:
- dependency-name: cdklabs-generative-ai-cdk-constructs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump mkdocstrings-python from 1.14.5 to 1.14.6 (#6050)

Bumps [mkdocstrings-python](https://github.com/mkdocstrings/python) from 1.14.5 to 1.14.6.
- [Release notes](https://github.com/mkdocstrings/python/releases)
- [Changelog](https://github.com/mkdocstrings/python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mkdocstrings/python/compare/1.14.5...1.14.6)

---
updated-dependencies:
- dependency-name: mkdocstrings-python
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps): bump squidfunk/mkdocs-material from `7e841df` to `c62453b` in /docs (#6052)

chore(deps): bump squidfunk/mkdocs-material in /docs

Bumps squidfunk/mkdocs-material from `7e841df` to `c62453b`.

---
updated-dependencies:
- dependency-name: squidfunk/mkdocs-material
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6055)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.14 to 1.36.16 (#6057)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.14 to 1.36.16.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6059)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a8 (#6061)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6062)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
feat(event_source): add class APIGatewayAuthorizerResponseWebSocket (#6058)

* Adding new class to WebSocket Authorizer

* Adding new class to WebSocket Authorizer

* Making mypy happy

* Making mypy happy

---------

Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps): bump zgosalvez/github-actions-ensure-sha-pinned-actions from 3.0.20 to 3.0.21 (#6064)

chore(deps): bump zgosalvez/github-actions-ensure-sha-pinned-actions

Bumps [zgosalvez/github-actions-ensure-sha-pinned-actions](https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions) from 3.0.20 to 3.0.21.
- [Release notes](https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions/releases)
- [Commits](https://github.com/zgosalvez/github-actions-ensure-sha-pinned-actions/compare/c3a2b64f69b7a1542a68f44d9edbd9ec3fc1455e...6eb1abde32fed00453b0d03497f4ba4fecba146d)

---
updated-dependencies:
- dependency-name: zgosalvez/github-actions-ensure-sha-pinned-actions
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump mkdocs-material from 9.6.2 to 9.6.3 (#6065)

Bumps [mkdocs-material](https://github.com/squidfunk/mkdocs-material) from 9.6.2 to 9.6.3.
- [Release notes](https://github.com/squidfunk/mkdocs-material/releases)
- [Changelog](https://github.com/squidfunk/mkdocs-material/blob/master/CHANGELOG)
- [Commits](https://github.com/squidfunk/mkdocs-material/compare/9.6.2...9.6.3)

---
updated-dependencies:
- dependency-name: mkdocs-material
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump coverage from 7.6.10 to 7.6.11 (#6067)

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.6.10 to 7.6.11.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.6.10...7.6.11)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

dependabot[bot] committed:
chore(deps-dev): bump ruff from 0.9.5 to 0.9.6 (#6066)

Bumps [ruff](https://github.com/astral-sh/ruff) from 0.9.5 to 0.9.6.
- [Release notes](https://github.com/astral-sh/ruff/releases)
- [Changelog](https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md)
- [Commits](https://github.com/astral-sh/ruff/compare/0.9.5...0.9.6)

---
updated-dependencies:
- dependency-name: ruff
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

anafalcao committed:
feat(metrics): disable metrics flush via environment variables (#6046)

* disable metrics env var cloudwatch

* add for datadog

* add tests

* acept none env var

* fix is disable metrics

* add documentation

* fix mypy

* add datadog tests

* disable when  powertools dev

* add disable to flush

* remove print

* Changing metrics flush decision

* Changing metrics flush decision

---------

Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): new pre-release 3.5.1a9 (#6069)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

anafalcao committed:
feat(data-masking): add custom mask functionalities (#5837)

* add custom mask functionalities

* change flags name to more intuitive

* fix type check error

* add draft documentation

* change doc examples

* style: format code with black

* fix format base

* add tests for new masks

* sub header for custom mask in docs

* masking rules to handle complex nest

* add test for masking rules

* modifications based on the feedback

* mypy and tests modification

* create more tests

* Refactoring tests

* Refactoring tests

* Refactoring tests

* Adding docstring + arg parameter

* Adding docstring + arg parameter

* Removing unnecessary code

* Removing unnecessary code

* Removing unnecessary code

---------

Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6070)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): bump version to 3.6.0 (#6071)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): layer docs update (#6072)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

leandrodamascena committed:
fix(security): fix encryption_context handling in data masking operations (#6074)

Enforcing encryption context

dependabot[bot] committed:
chore(deps-dev): bump boto3-stubs from 1.36.16 to 1.36.17 (#6078)

Bumps [boto3-stubs](https://github.com/youtype/mypy_boto3_builder) from 1.36.16 to 1.36.17.
- [Release notes](https://github.com/youtype/mypy_boto3_builder/releases)
- [Commits](https://github.com/youtype/mypy_boto3_builder/commits)

---
updated-dependencies:
- dependency-name: boto3-stubs
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump mkdocstrings-python from 1.14.6 to 1.15.0 (#6079)

Bumps [mkdocstrings-python](https://github.com/mkdocstrings/python) from 1.14.6 to 1.15.0.
- [Release notes](https://github.com/mkdocstrings/python/releases)
- [Changelog](https://github.com/mkdocstrings/python/blob/main/CHANGELOG.md)
- [Commits](https://github.com/mkdocstrings/python/compare/1.14.6...1.15.0)

---
updated-dependencies:
- dependency-name: mkdocstrings-python
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

dependabot[bot] committed:
chore(deps-dev): bump coverage from 7.6.11 to 7.6.12 (#6080)

Bumps [coverage](https://github.com/nedbat/coveragepy) from 7.6.11 to 7.6.12.
- [Release notes](https://github.com/nedbat/coveragepy/releases)
- [Changelog](https://github.com/nedbat/coveragepy/blob/master/CHANGES.rst)
- [Commits](https://github.com/nedbat/coveragepy/compare/7.6.11...7.6.12)

---
updated-dependencies:
- dependency-name: coverage
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump aws-actions/configure-aws-credentials from 4.0.3 to 4.1.0 (#6082)

chore(deps): bump aws-actions/configure-aws-credentials

Bumps [aws-actions/configure-aws-credentials](https://github.com/aws-actions/configure-aws-credentials) from 4.0.3 to 4.1.0.
- [Release notes](https://github.com/aws-actions/configure-aws-credentials/releases)
- [Changelog](https://github.com/aws-actions/configure-aws-credentials/blob/main/CHANGELOG.md)
- [Commits](https://github.com/aws-actions/configure-aws-credentials/compare/4fc4975a852c8cd99761e2de1f4ba73402e44dd9...ececac1a45f3b08a01d2dd070d28d111c5fe6722)

---
updated-dependencies:
- dependency-name: aws-actions/configure-aws-credentials
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Ana Falcão <anafalcao@poli.ufrj.br>

github-actions[bot] committed:
chore(ci): new pre-release 3.6.1a0 (#6084)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6085)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>

leandrodamascena committed:
docs(roadmap): update roadmap (#6077)

* Adding roadmap

* Roadmap

dependabot[bot] committed:
chore(deps): bump squidfunk/mkdocs-material from `c62453b` to `f5bcec4` in /docs (#6087)

chore(deps): bump squidfunk/mkdocs-material in /docs

Bumps squidfunk/mkdocs-material from `c62453b` to `f5bcec4`.

---
updated-dependencies:
- dependency-name: squidfunk/mkdocs-material
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump aws-cdk from 2.178.1 to 2.178.2 (#6089)

Bumps [aws-cdk](https://github.com/aws/aws-cdk/tree/HEAD/packages/aws-cdk) from 2.178.1 to 2.178.2.
- [Release notes](https://github.com/aws/aws-cdk/releases)
- [Changelog](https://github.com/aws/aws-cdk/blob/v2.178.2/CHANGELOG.v2.md)
- [Commits](https://github.com/aws/aws-cdk/commits/v2.178.2/packages/aws-cdk)

---
updated-dependencies:
- dependency-name: aws-cdk
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): new pre-release 3.6.1a1 (#6090)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

github-actions[bot] committed:
chore(ci): changelog rebuild (#6091)

Co-authored-by: Powertools for AWS Lambda (Python) bot <151832416+aws-powertools-bot@users.noreply.github.com>
Co-authored-by: Leandro Damascena <lcdama@amazon.pt>

sinofseven committed:
add test for s3.py

sinofseven committed:
add test for secrets_manager.py

sinofseven committed:
Merge branch 'develop' into impl-prototype-async-execution-resolver

leandrodamascena commented:
Hey @sinofseven! We're closing this PR for now, as we need to write an RFC to get more customer feedback! This is still on our radar for 2026, and once we have more data and visibility on this feature, I'll mention you for review - if you're interested - and give you proper credit when we release it.

Thanks for all the efforts, I really appreciate it.
